### PR TITLE
Add parentheses to arrow functions single parameters (part 7)

### PR DESCRIPTION
**Pull request by teoli2003**
See #18677

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/promise/all/index.md [hunk 1]
```diff
@@ -233,9 +233,9 @@ const p2 = new Promise((resolve, reject) => {
 });
 
 Promise.all([
-  p1.catch(error => { return error }),
-  p2.catch(error => { return error }),
-]).then(values => {
+  p1.catch((error) => { return error }),
+  p2.catch((error) => { return error }),
```
Comment (lionralfs) [inline files/en-us/web/javascript/reference/global_objects/promise/all/index.md:237]:
```suggestion
  p1.catch((error) => error),
  p2.catch((error) => error),
```

File: files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md [hunk 2]
```diff
@@ -306,9 +306,9 @@ computation(console.log, 5) // -> 14
 const computation2 = (input, callback) => {
   const f6 = x=> div4(callback, x);
```
Comment (lionralfs) [inline files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md:307]:
```suggestion
  const f6 = x => div4(callback, x);
```

File: files/en-us/web/javascript/reference/global_objects/promise/index.md [hunk 3]
```diff
@@ -83,12 +83,12 @@ Using {{JSxRef("Functions/Arrow_functions", "Arrow Function Expressions", "", 1)
...
-  .then(value => { return value + ' and bar'; })
-  .then(value => { return value + ' and bar again'; })
-  .then(value => { return value + ' and again'; })
-  .then(value => { return value + ' and again'; })
-  .then(value => { console.log(value) })
-  .catch(err => { console.log(err) });
+  .then((value) => { return value + ' and bar'; })
+  .then((value) => { return value + ' and bar again'; })
+  .then((value) => { return value + ' and again'; })
+  .then((value) => { return value + ' and again'; })
```
Comment (lionralfs) [inline files/en-us/web/javascript/reference/global_objects/promise/index.md:89]:
```suggestion
  .then((value) => `${value} and bar`)
  .then((value) => `${value} and bar again`)
  .then((value) => `${value} and again`)
  .then((value) => `${value} and again`)
```

File: files/en-us/web/javascript/reference/global_objects/promise/index.md [hunk 4]
```diff
@@ -83,12 +83,12 @@ Using {{JSxRef("Functions/Arrow_functions", "Arrow Function Expressions", "", 1)
...
+  .then((value) => { console.log(value) })
+  .catch((err) => { console.log(err) });
```
Comment (lionralfs) [inline files/en-us/web/javascript/reference/global_objects/promise/index.md:91]:
```suggestion
  .catch((err) => { console.error(err) });
```

File: files/en-us/web/javascript/reference/global_objects/promise/then/index.md [hunk 5]
```diff
@@ -217,7 +217,7 @@ even though the previous Promise in the chain was rejected.
 ```js
 Promise.reject()
   .then(() => 99, () => 42) // onRejected returns 42 which is wrapped in a resolving Promise
-  .then(solution => console.log('Resolved with ' + solution)); // Resolved with 42
+  .then((solution) => console.log('Resolved with ' + solution)); // Resolved with 42
```
Comment (lionralfs) [inline files/en-us/web/javascript/reference/global_objects/promise/then/index.md:220]:
```suggestion
  .then((solution) => console.log(`Resolved with ${solution}`)); // Resolved with 42
```

File: files/en-us/web/javascript/reference/global_objects/promise/then/index.md [hunk 6]
```diff
@@ -229,7 +229,7 @@ Promise.resolve()
     // Makes .then() return a rejected promise
     throw new Error('Oh no!');
   })
-  .catch(error => {
+  .catch((error) => {
     console.error('onRejected function called: ' + error.message);
```
Comment (lionralfs) [inline files/en-us/web/javascript/reference/global_objects/promise/then/index.md:233]:
```suggestion
    console.error(`onRejected function called: ${error.message}`);
```

File: files/en-us/web/javascript/reference/global_objects/promise/then/index.md [hunk 7]
```diff
@@ -205,7 +205,7 @@ Promise.resolve()
   })
   .then(() => {
     console.log('Not called.');
-  }, error => {
+  }, (error) => {
     console.error('onRejected function called: ' + error.message);
```
Comment (lionralfs) [inline files/en-us/web/javascript/reference/global_objects/promise/then/index.md:209]:
```suggestion
    console.error(`onRejected function called: ${error.message}`);
```

File: files/en-us/web/javascript/reference/global_objects/regexp/@@split/index.md [hunk 8]
```diff
@@ -79,7 +79,7 @@ modify the default behavior.
 class MyRegExp extends RegExp {
   [Symbol.split](str, limit) {
     let result = RegExp.prototype[Symbol.split].call(this, str, limit);
-    return result.map(x => "(" + x + ")");
+    return result.map((x) => "(" + x + ")");
```
Comment (lionralfs) [inline files/en-us/web/javascript/reference/global_objects/regexp/@@split/index.md:82]:
```suggestion
    return result.map((x) => `(${x})`);
```

**Review by Josh-Cena — APPROVED** [review]
LGTM

**Comment by lionralfs** [comment]
Bad timing 😁

File: files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md (hunk 2 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md:307]: Yeah @teoli2003 I suspect there are more cases. Maybe widen the search?

Comment (lionralfs) [reply files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md:307]: Woops, should of course be:
```suggestion
  const f6 = (x) => div4(callback, x);
```

**Comment by Josh-Cena** [comment]
@lionralfs the other cases seem to be for prefer-template and arrow-body-style. Would love to see batch PRs for those!

**Comment by lionralfs** [comment]
As they're outside `web/api` we'd have to fix syntax errors first I think

File: files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md (hunk 2 above)
Comment (teoli2003) [reply files/en-us/web/javascript/reference/global_objects/array/reduceright/index.md:307]: Yes, I did make the space mandatory. I will make another pass to remove `_`, and cases without space.

**Pull request merged**